android/filament-android/src/main/java/com/google/android/filament/VertexBuffer.java

```diff
@@ -197,7 +397,27 @@ public void setBufferAt(@NonNull Engine engine, int bufferIndex, @NonNull Buffer
             throw new BufferOverflowException();
         }
     }
-
+    /**
+     * Convenience function that consumes normal vectors (and, optionally, tangent vectors) and
+     * produces quaternions that can be passed into a TANGENTS buffer.
+     *
+     * <p>
+     * The given output buffer must be preallocated with at least quatCount * outStride bytes.
+     * <p>
+     *
+     * <p>
+     * Normals are required but tangents are optional, in which case this function tries to generate
+     * reasonable tangents. The given normals should be unit length.
+     * </p>
+     *
+     * <p>
+     * If supplied, the tangent vectors should be unit length and should be orthogonal to the
+     * normals. The w component of the tangent is a sign (-1 or +1) indicating handedness of the
+     * basis.
+     * </p>
+     *
```

Comment:
Yeah we can't @deprecate this yet because we don't have JNI for SurfaceOrientation.  Thanks for doing this!

Reply:
Yup, I noticed that. When we do add JNI support, we can add the @deprecation tag then.